src/Microsoft.PowerShell.Commands.Utility/commands/utility/WaitEventCommand.cs

```diff
@@ -84,13 +102,13 @@ protected override void ProcessRecord()
 
             while (!received)
             {
-                if (_timeoutInSeconds >= 0)
+                if (_timeoutTimespan.TotalMilliseconds >= 0)
                 {
-                    if ((DateTime.UtcNow - startTime).TotalSeconds > _timeoutInSeconds)
+                    if ((DateTime.UtcNow - startTime) > _timeoutTimespan)
```

Comment:
We could use new static Stopwatch.[GetElapsedTime](https://source.dot.net/System.Private.CoreLib/R/221a9d2c840277cc.html)